src/System.Devices.Gpio/System.Devices.Gpio.Samples/Bme280.cs

```diff
@@ -253,41 +288,62 @@ private void ReadSensorCoefficients()
             _calibrationData.dig_H6 = (sbyte)Read8(BME280_DIG_H6_REG);
         }
 
-        /// <summary>
-        /// Transfers data over the SPI bus
-        /// </summary>
-        private byte SpiTransfer(byte x = 0)
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private byte Read()
         {
-            var readBuffer = new byte[1];
+            var buffer = new byte[1];
 
-            if (x == 0)
-            {
-                _spiDevice.Read(readBuffer);
-            }
-            else
+            switch (_protocol)
             {
-                var writeBuffer = new byte[] { x };
+                case ConnectionProtocol.Spi:
+                    _spiDevice.Read(buffer);
+                    break;
+
+                case ConnectionProtocol.I2c:
+                    _i2cDevice.Read(buffer);
+                    break;
 
-                _spiDevice.TransferFullDuplex(writeBuffer, readBuffer);
+                default:
+                    throw new NotSupportedException($"Connection protocol '{_protocol}' not supported");
             }
 
-            byte result = readBuffer[0];
+            byte result = buffer[0];
             return result;
         }
 
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private void Write(byte x)
+        {
+            var buffer = new byte[] { x };
```

Comment:
We shouldn't have to allocate on every Write call.  Is it possible to refactor the internals so you can avoid this new byte[]{x}?

Reply:
Yes, I will change it to reuse always the same buffer.